src/iso19111/io.cpp

```diff
@@ -6424,6 +6424,46 @@ static BaseObjectNNPtr createFromUserInput(const std::string &text,
             }
             throw;
         }
+    } else if (tokens.size() == 3) {
+        // ESRI:103668+EPSG:5703 ... compound
+        auto tokensCenter = split(tokens[1], '+');
+        if (tokensCenter.size() == 2) {
+            if (!dbContext) {
+                throw ParsingException("no database context specified");
+            }
+            DatabaseContextNNPtr dbContextNNPtr(NN_NO_CHECK(dbContext));
+
+            const auto &authName1 = tokens[0];
+            const auto &code1 = tokensCenter[0];
+            const auto &authName2 = tokensCenter[1];
+            const auto &code2 = tokens[2];
+
+            crs::CRSPtr crs1 = nullptr;
+            crs::CRSPtr crs2 = nullptr;
+
+            const auto authorities = dbContextNNPtr->getAuthorities();
+            for (const auto &authCandidate : authorities) {
+                if (ci_equal(authCandidate, authName1)) {
+                    auto factory =
+                        AuthorityFactory::create(dbContextNNPtr, authCandidate);
+                    crs1 =
```

Comment:
some compilers require an explict .as_nullable() to cast from a non-null shared pointed to a regular one. But here the best option would probably be to just assign the real authority names to a std::string and then do like the tokens.size() == 2 case avoid the non-null test on crs1 / crs2.
Actually you could skip the ci_equal() test and just use what the user provides. The case-insensitive comparison in the tokens.size() == 2 case was done because of legacy practice of using "epsg:XXXX", but here for a new supported syntax we can be stricter. That will make the code simpler

Reply:
Thanks. Without `ci_equal` the code is much simpler.

Reply:
FYI, `projinfo esri:103668+epsg:5703` does work perfectly. However `projinfo ogc:CRS84+EPSG:5773` fails. (something that I already noticed with `--list-crs`)

`AuthorityFactory::create` does a `ci_equal` for `EPSG`, `ESRI` and `PROJ`.